Mathlib/Analysis/Convex/StrictConvexBetween.lean

```diff
@@ -78,3 +91,24 @@ theorem Collinear.sbtw_of_dist_eq_of_dist_lt {p p₁ p₂ p₃ : P} {r : ℝ}
   · rintro rfl
     exact hp₂.ne hp₃
 #align collinear.sbtw_of_dist_eq_of_dist_lt Collinear.sbtw_of_dist_eq_of_dist_lt
+
+end PseudoMetricSpace
+
+section MetricSpace
+variable [MetricSpace P] [NormedAddTorsor V P] {a b c : P}
+
+/-- If the triangle `abc` is flat (the triangle inequality is an equality), then `b` lies between
+`a` and `c`.
+
+TODO: Deduplicate from `dist_add_dist_eq_iff`. -/
+lemma dist_add_dist_eq_iff_wbtw : dist a b + dist b c = dist a c ↔ Wbtw ℝ a b c := by
+  obtain rfl | hac := eq_or_ne a c
+  · simp [dist_comm, eq_comm]
+  refine ⟨fun habc ↦ ?_, Wbtw.dist_add_dist⟩
+  refine ⟨dist a b / dist a c, ⟨by positivity, div_le_one_of_le ?_ <| by positivity⟩, Eq.symm <|
+    eq_lineMap_of_dist_eq_mul_of_dist_eq_mul (div_mul_cancel _ <| dist_ne_zero.2 hac).symm ?_⟩
+  · rw [←habc]
+    exact le_add_of_nonneg_right <| by positivity
+  · rwa [one_sub_mul, div_mul_cancel _ <| dist_ne_zero.2 hac, eq_sub_iff_add_eq']
```

Comment:
I had a slightly shorter proof of this in a branch somewhere,
```
theorem dist_add_dist_eq_iff' (x y z : P) : dist x y + dist y z = dist x z ↔ Wbtw ℝ x y z := by
  have :
      dist (x -ᵥ x) (y -ᵥ x) + dist (y -ᵥ x) (z -ᵥ x) = dist (x -ᵥ x) (z -ᵥ x) ↔
        y -ᵥ x ∈ segment ℝ (x -ᵥ x) (z -ᵥ x) := by
    rw [dist_add_dist_eq_iff]
  simp_rw [dist_vsub_cancel_right, ← affineSegment_eq_segment, ← affineSegment_vsub_const_image] at this 
  rwa [(vsub_left_injective _).mem_set_image] at this
```
but maybe you deliberately didn't want to go via `dist_add_dist_eq_iff`?

Reply:
Yes, because `dist_add_dist_eq_iff` is a less general theorem that will eventually be subsumed. Of course the answer is that we should have a tactic that takes any normed torsor statement, discharges the case where the normed torsor is empty automatically and turns the statement into a normed space one when the normed torsor is nonempty.

But we don't have that. Maybe we could whine about it to @JLimperg?